### DISCUSSION TITLE
jvm/oom

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#291] [Announcements]
# jvm/oom

本文介绍了一次内存溢出的排查优化实战，彻底干掉臭名昭著的 OOM。

https://javabetter.cn/jvm/oom.html

**Comment by chaofengdev** [comment]
一次很好的实战经验分享。但是依旧没看懂。

**Comment by SetUpLi** [comment]
Disruptor配置不当：
		预分配内存直接进入老年代 10GB  =  bufferSize（1024*1024）  × 单个事件对象大小（10K）
		浮动垃圾无处安放：业务产生的临时对象在年轻代熬过几次 GC 后想进入老年代，老年代业务对象越来越多，发现老年代已经被 Disruptor 塞满了。触发频繁的 Full GC，FUll GC只能回收部分业务对象，无法回收Disruptor预分配对象 最终因无法晋升而 OOM。
